File: packages/shared/src/membership.ts
import { MembershipStatus, Role } from './enums';

/**
 * The role/status/joined attributes for a new membership, independent of the
 * account/organization identifiers the persistence layer attaches.
 */
export interface CreatorMembershipInput {
  role: Role;
  status: MembershipStatus;
  joinedAt: Date;
}

/**
 * The membership granted to the person who creates an organization at signup
 * (spec 01, requirements 5–6): the creator is the organization's first `admin`
 * with an `active` membership. There is no separate "owner" concept.
 *
 * @param joinedAt when the membership was created (defaults to now).
 */
export function createAdminMembershipInput(joinedAt: Date = new Date()): CreatorMembershipInput {
  return {
    role: Role.Admin,
    status: MembershipStatus.Active,
    joinedAt,
  };
}

File: src/Devscribed.Admin.Web/Models/Membership.cs
namespace Devscribed.Admin.Web.Models;

public class Membership
{
    public Guid Id { get; set; }
    public Guid AccountId { get; set; }
    public Guid OrganizationId { get; set; }
    public string Role { get; set; } = "admin";
    public string Status { get; set; } = "active";
    public DateTime JoinedAt { get; set; }

    public Account Account { get; set; } = null!;
    public Organization Organization { get; set; } = null!;
}
